subsys/bluetooth/audio/ascs.c

```diff
@@ -279,15 +279,23 @@ static void ase_enter_state_qos_configured(struct bt_ascs_ase *ase)
 {
 	struct bt_bap_stream *stream = ase->ep.stream;
 	struct bt_bap_stream_ops *ops;
+	const enum bt_bap_ep_state ep_state = ascs_ep_get_state(&ase->ep);
 
 	__ASSERT_NO_MSG(stream != NULL);
 
+	ops = stream->ops;
+	if ((ep_state == BT_BAP_EP_STATE_ENABLING) || (ep_state == BT_BAP_EP_STATE_STREAMING)) {
+		if (ops != NULL && ops->disabled != NULL) {
+			ops->disabled(stream);
+		}
+	}
```

Comment:
`ase_enter_state_qos_configured` is called by `state_transition_work_handler` which does `ase->ep.status.state = new_state;` so `ep_state` will always be `BT_BAP_EP_STATE_QOS_CONFIGURED` in this function, so this change won't affect anything. 

To verify the check, the unit tests for ASCS should be expanded to verify that the `disabled` callback is called. Maybe ideally you can modify the test first and then use that to verify the fix (Test driven development).